### ISSUE TITLE
Typescript Type defintions not being pulled in

### ISSUE THREAD

**Issue by GimpMaster** [agenda/agenda#1424]
I just recently upgraded from Agenda 3.1.0 to 4.2.1

I'm using typescript and the documentation says to change from 

`import Agenda from "agenda";`

to

``import { Agenda } from "agenda/es";`

While this does work I'm finding that typescript and my IDE, treats this as if Agenda is **"any"**. It no longer provides any type checking on the core Agenda object.

Do I still need to provide the definetly typed "@types/agenda" module? Npm says its fully TS.

**Comment by deni-begaj** [comment]
Wondering the same thing. From their README is mentioned to use `agenda/es` import.

But I think that is oriented toward module/ES projects.

For CJS (usual) projects we may still need to use `require`.

**Comment by GimpMaster** [comment]
Yeah I just went back to 3.1.0 with the old @types. I couldn't figure out how to get a Jetbrains IDE to use the correct types with autocomplete. They just treated the Agenda object as an any which gave no type completion or checking.

**Comment by simllll** [comment]
You could try https://github.com/hokify/agenda types are working there correctly

**Comment by GimpMaster** [comment]
@simllll - Are you forking and trying to pull request back into this repo or keeping them separate?

Or maybe this isn't your repo.

**Comment by simllll** [comment]
@GimpMaster I was trying that the the beginning of the fork, but changes got too big and there where some breaking changes. Unfortunately for me I have too less time to bring them back here and spend time on all the discussions that are sometimes needed for that, but feel free to bring changes back to here! So to actually answer your question: my plan is to keeping them seperate at least from my side, but willing to help and review anything that can lead to a situation where this agenda repo will get a decent state again ;-)

**Comment by GimpMaster** [comment]
Thank you. I'll give your repo a test go in a scratch project here in the next week or so.
